### DISCUSSION TITLE
ROM dump metadata

### DISCUSSION THREAD

**Discussion by mariomadproductions** [mamedev/discussions#46] [General]
I think it would be a good idea if MAME stored ROM dump metadata in a machine readable way, like No-Intro and (to some degree) Redump do.

E.g. store the following info for each dump:
* Photos
* Serials for each physical item that was dumped
* Dump tool
* Dump creator
* Dump date

(i.e. provenance info)

This would enable the following:
* Checking/correcting transcriptions using photos
* Knowing what exact physical item has been dumped/what it looks like
* Knowing who to ask if a physical item needs to be redumped
* Knowing if something should be redumped due to a dump tool being later discovered to be faulty
* Knowing if a dump has been verified/redumped

These things have been very useful for projects like No-Intro and Redump, and from what I can see, similar tasks are attempted in MAME (and any software digitisation project).

I'd be interested to hear peoples' thoughts on this.

**Comment by galibert** [comment]
I doubt a catalogue of copyright violations and people responsible for them is something we want to have in Mame.

**Reply by mariomadproductions** [reply]
What do you mean? The MAME website solicits ROM dumps and [this](https://wiki.mamedev.org/index.php/FAQ:ROMs) FAQ page says making ROM dumps is a valid way to use MAME. Is physical hardware preservation and listing metadata not legal, and a core part of what MAME and other preservation groups (including professional organisations) do? Also the dumper _nicknames_ (which is all that is needed) are already recorded in the changelogs, just not in a machine readable way.

**Reply by rb6502** [reply]
Making ROM dumps for your personal use is a valid and legal way to use MAME.  You're discussing a list of the people who made the dumps that are being distributed, which is a very different thing.

Also, it wouldn't have the benefits you think.  Most boards dumped for MAME do not belong to the dumper and they only have possession for long enough to dump them, at which point they're often sold or otherwise disappear.

**Reply by mariomadproductions** [reply]
Being distributed != they distributed them, since the dumps could just be verifications of stuff floating about on the web. But again, dumper names are apparently aleady stored in MAME, just not in a very machine readable way.

That's fair, although the other metadata items would still apply.

**Reply by ghost** [reply]
> Being distributed != they distributed them, since the dumps could just be verifications of stuff floating about on the web. But again, dumper names are apparently aleady stored in MAME, just not in a very machine readable way.
> 
> That's fair, although the other metadata items would still apply.

It's useless to talk to these guys :)
They ALWAYS think they know it better.
You are of course right, but please look at my answer for the REAL reason this process can't be documented...

**Comment by cuavas** [comment]
We don’t have the resources to take on a project like cataloguing PCB pictures, etc.  We’re stretched thin on the ground with the stuff we already do.  If you want a project that does this, gather a community of like-minded people and do it.  Don’t try to push it as scope creep for an existing project.

**Reply by mariomadproductions** [reply]
Yeah, may not be a suitable thing to go in the MAME repo.

**Reply by ghost** [reply]
> We don’t have the resources to take on a project like cataloguing PCB pictures, etc. We’re stretched thin on the ground with the stuff we already do. If you want a project that does this, gather a community of like-minded people and do it. Don’t try to push it as scope creep for an existing project.

You guys take some ROMs from websites, split them in parts and then call them "MAME" ROMs. Of course you can't document the process where they come from... They are not dumped by the MAME team (especially for the softlist ROMs)

The proof is the tool ucon64 applied to many of the Software List ROMs. You forget 1 thing: after splitting , the internal CRC checksum is still in the ROM and ucon64 of course can't verify it anymore (since you guys split the ROM).

WAY TO GO!!! :)
This is what they call "software preservation" ;-)
This is the REAL reason this is closed here...

For interested people: https://ucon64.sourceforge.io/
